Update PX4 Porting Guide organization for clarity

joshuawhitehead opened the pull request:
Attempted to group information in a more consistent (and easily expandable) manner. Updated language in various locations to add clarity and be more consistent within the porting guide and with respect to other documentation files.  Added links and references in a few locations to other PX4 documentation as appropriate.  Corrected multiple dead links.

joshuawhitehead committed:
Update porting-guide.md organization for clarity

Attempted to group information in a more consistent (and easily expandable) manner. Also updated language in various locations to add clarity and be more consistent within the porting guide and with respect to other documentation files.

hamishwillee commented on en/debug/porting-guide.md:
```diff
@@ -2,55 +2,67 @@
+## PX4 Architecture
 
-PX4 consists of two main layers: The board support and middleware layer on top of the host OS (NuttX, Linux or any other POSIX platform like Mac OS). And the applications (Flight Stack in [src/modules](https://github.com/PX4/Firmware/tree/master/src/modules)\).
+PX4 consists of two main layers: The [board support and middleware layer](https://dev.px4.io/en/middleware/) on top of the host OS (NuttX, Linux or any other POSIX platform like Mac OS), and the applications (Flight Stack in [src/modules](https://github.com/PX4/Firmware/tree/master/src/modules)\).  Please reference the [PX4 Architectural Overview](../concept/architecture.md) for more information.
```
Links within the same doc should be relative. So for example this `https://dev.px4.io/en/middleware/` should be `../middleware/README.md`

joshuawhitehead replied:
Ah, ok, so anything that's in the "Devguide" repo should use a relative link, I should only use a full URL if I need to reference something else, e.g. the User Guide?  I was working through the GitHub interface to make this update and was a little unsure of how it handled things.  I will update this.

hamishwillee replied:
That is correct. You've done it right throughout pretty much - except for this one.

hamishwillee commented on en/debug/porting-guide.md:
```diff
@@ -2,55 +2,67 @@
-* Reference config: Running `make qurt_eagle_release` builds the Snapdragon Flight reference config.
+The configuration files for NuttX based boards, including linker scripts and other required settings are located under [platforms/nuttx/nuttx-configs](https://github.com/PX4/Firmware/tree/master/platforms/nuttx/nuttx-configs).
+
+The following example uses FMUv5 as it is the current [reference configuration](./reference-design.md) for NuttX based flight controllers:
```
This link will work, as will `(reference-design.md)`. I have a slight preference for the full relative path `(../debug/reference-design.md)` as it reminds me this file needs to be moved to somewhere with a better name!

joshuawhitehead replied:
Agreed, that makes sense, I will update accordingly.

hamishwillee replied:
Thanks!

hamishwillee commented on en/debug/porting-guide.md:
```diff
@@ -2,55 +2,67 @@
+* The start script is located in [posix-configs/](https://github.com/PX4/Firmware/tree/master/posix-configs).
+* The OS configuration is part of the default Linux image (TODO: Provide location of LINUX IMAGE and flash instructions).
+* The PX4 middleware configuration is located in [src/drivers/boards](https://github.com/PX4/Firmware/tree/master/src/drivers/boards). TODO: ADD BUS CONFIG
+* Drivers: [DriverFramework](https://github.com/px4/DriverFramework).
```
This is true, though in future these are expected to use same drivers as NuttX

joshuawhitehead replied:
Would it be worth adding a "Future Plans" subsection to add things like that? In the chance you pique someone's interest to help make it happen :-)

hamishwillee replied:
Not for now. Daniel Agar is going to do this particular task and I don't know of other future plans. If you know of enough to justify them, then definitely.

hamishwillee commented on en/debug/porting-guide.md:
```diff
@@ -2,55 +2,67 @@
-* Reference config: Running `make qurt_eagle_release` builds the Snapdragon Flight reference config.
+The configuration files for NuttX based boards, including linker scripts and other required settings are located under [platforms/nuttx/nuttx-configs](https://github.com/PX4/Firmware/tree/master/platforms/nuttx/nuttx-configs).
+
+The following example uses FMUv5 as it is the current [reference configuration](./reference-design.md) for NuttX based flight controllers:
```
Calling FMUv5 the current reference config for NuttX based flight controllers feels a little odd because historically the variants are not necessarily sequential - ie FMU versions only really infer binary compatibility and a lower number could be the same level of advanced. I guess it is the most recent and the one we would hope manufacturers would target so leave as is.

joshuawhitehead replied:
Understood about the binary compatibility/feature set, I had read that in other places as well, I had intended it only as you mentioned- If someone is porting to a brand new board, FMUv5 would be the best place to start.  I thought I had seen a similar sentence in the reference-design.md doc, but now I'm not... I will add a note here to remind people that the versions/generations to not correlate with capabilities, simply that v5, being the most recent version, reflects the most up to date methods for hardware and code structure.

hamishwillee replied:
I actually wouldn't add a note here (maybe back in the reference-design doc though!). 
What I'd do is change as shown:

FROM
> The following example uses FMUv5 as it is the current [reference configuration](./reference-design.md) for NuttX based flight controllers:

TO

> The following example uses FMUv5 as it is a recent [reference configuration](./reference-design.md) for NuttX based flight controllers:

dagar commented on en/debug/porting-guide.md:
```diff
@@ -2,55 +2,67 @@
+* The OS configuration is part of the default Linux image (TODO: Provide location of LINUX IMAGE and flash instructions).
+* The PX4 middleware configuration is located in [src/drivers/boards](https://github.com/PX4/Firmware/tree/master/src/drivers/boards). TODO: ADD BUS CONFIG
+* Drivers: [DriverFramework](https://github.com/px4/DriverFramework).
+* Reference config: Running `make qurt_eagle_release` builds the Snapdragon Flight reference config.
```
This target isn't valid.

hamishwillee replied:
@joshuawhitehead Can you fix this up too  - it should just be `make eagle_default` (according to [this](https://dev.px4.io/en/setup/building_px4.html#qurt--snapdragon-based-boards).

joshuawhitehead replied:
Will do.  This was copy-paste from the current version of the porting guide, so the target may have changed since that was originally written.

joshuawhitehead committed:
Update porting-guide per review feedback

Addressed specific changes requested during feedback of initial patch.  No other changes or additions were made.
